frontend/src/AutocompleteWrapper.ts

```diff
@@ -1,206 +1,144 @@
-
-      const autocompleteParams = {
-        hint: false,
-        debug,
-        templates: this.getDropdownTemplates(poweredBy),
-        appendTo: 'body',
-      };
-
-      const searchParams = {
-        analytics,
-        hitsPerPage,
-        highlightPreTag: '<span class="aa-hit--highlight">',
-        highlightPostTag: '</span>',
-        attributesToSnippet: [
-          `description:${nbSnippetWords}`,
-          `content:${nbSnippetWords}`,
-        ],
-        snippetEllipsisText: '...',
-      };
-
-      const sources = [
-        {
-          name: 'hits',
-          source: this.createSource(searchParams),
-          templates: {
-            suggestion: this.createRenderSuggestion(sizeModifier),
-          },
-        },
-      ];
-
-      const aa: AutocompleteJs = autocomplete(
-        $input,
-        autocompleteParams,
-        sources
-      );
-
-      const handleSelected = this.createHandleSelected();
-      aa.on('autocomplete:selected', handleSelected);
-
-      return aa;
+    const instance = autocomplete<AlgoliaRecord>({
+      container: $input,
+      autoFocus: false,
+      placeholder: 'Search...',
+      debug: this.options.debug,
+      openOnFocus: true,
+      panelPlacement: 'input-wrapper-width',
+      classNames: {
+        sourceFooter: 'aa-powered-by',
+      },
+      getSources: () => {
+        return [this.getSources()];
+      },
     });
+    this.applyTheme($input.firstElementChild as HTMLElement);
 
-    // Store debug variables
-    this.$inputs = $inputs;
-    this.autocompletes = autocompletes;
+    this.autocomplete = instance;
   }
 
-  private computeIndexName(siteId: string, branch: string): string {
+  private computeIndexName(): string {
+    const { siteId, branch } = this.options;
+
     // Keep in sync with crawler code in /netlify/crawl
     const cleanBranch = branch
       .replace(/[^\p{L}\p{N}_.-]+/gu, '-')
       .replace(/-{2,}/g, '-');
     return `netlify_${siteId}_${cleanBranch}_all`;
   }
 
-  private createClient(appId: string, apiKey: string): SearchClient {
-    const client = algoliasearch(appId, apiKey);
+  private createClient(): SearchClient {
+    const client = algoliasearch(this.options.appId, this.options.apiKey);
     client.addAlgoliaAgent(`Netlify integration ${version}`);
     return client;
   }
 
-  private getInputs(inputSelector: string): HTMLInputElement[] {
-    return Array.from(document.querySelectorAll(inputSelector));
-  }
-
-  private computeSizeModifier(inputWidth: number): SizeModifier | null {
-    if (inputWidth < XS_WIDTH) return 'xs';
-    if (inputWidth < SM_WIDTH) return 'sm';
-    return null;
-  }
-
-  private computeNbSnippetWords(inputWidth: number): number {
-    if (inputWidth < XS_WIDTH) return 0;
-    if (inputWidth < SM_WIDTH) return 3 + Math.floor(inputWidth / 35);
-    return Math.floor(inputWidth / 20);
-  }
-
-  private getDropdownTemplates(poweredBy: boolean): { footer?: string } {
-    if (!poweredBy) return {};
```

Comment:
This breaks the `poweredBy` option which is not taken into account anymore

Reply:
good catch